os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c

```diff
@@ -2827,4 +2831,112 @@ int wifi_set_null1_param(uint8_t check_period, uint8_t limit, uint8_t interval)
 }
 #endif
 
+#include "freertos/wrapper.h"
+extern Rltk_wlan_t rltk_wlan_info[NET_IF_NUM];
+
+typedef struct WIFI_MONITOR{
+	/*RX related*/
+	unsigned long rx_success_pkt_count;
+	unsigned long rx_success_bytes_count;
+	unsigned long rx_dropped_pkt_count;	
+	int rx_rssi;
+	u32 rxd_beacon_count;	
+	
+	/*TX related*/
+	unsigned long tx_success_pkt_count;
+	unsigned long tx_success_bytes_count;	
+	unsigned long tx_dropped_pkt_count;
+}wifi_monitor;
+
+wifi_monitor wifi_monitor_info = {0}; 
+wifi_monitor wifi_monitor_prev = {0};
+
+void wifi_get_monitor_info(unsigned char idx)
+{
+	struct net_device *dev = NULL;
+	struct net_device_stats *stats;
+	int rssi = 0;
+	u32 total_beacon_count = 0;
+	u32 rxd_beacon_count = 0;	
+	u32 tmp_rxd_beacon_count = 0;
+	u32 missed_beacon_count = 0;
+
+	save_and_cli();
+	
+	if(rltk_wlan_info[idx].enable) {
+		/*to get TX & RX packets status*/
+		dev = rltk_wlan_info[idx].dev;
+		stats = dev->get_stats(dev);
+		wifi_monitor_info.rx_success_pkt_count = stats->rx_packets;
+		wifi_monitor_info.rx_success_bytes_count = stats->rx_bytes;
+		wifi_monitor_info.rx_dropped_pkt_count = stats->rx_dropped;
+		
+		wifi_monitor_info.tx_success_pkt_count = stats->tx_packets;
+		wifi_monitor_info.tx_success_bytes_count = stats->tx_bytes;
+		wifi_monitor_info.tx_dropped_pkt_count = stats->tx_dropped;
+		
+		/*to get average rssi*/
+		wifi_get_rssi(&rssi);
+		wifi_monitor_info.rx_rssi = rssi;
+		
+		/*to get beacon miss count*/
+		if (wifi_is_connected_to_ap() == RTW_SUCCESS)
+			total_beacon_count = (WIFI_MONITOR_INFO_TIMER)/100;		/*beacon interval: 100 ms*/

```

Comment:
can a system get exact total_beacon_count by this? this depends on that beacon interval is accurate and exact

Reply:
There is no API to get the total beacon count directly. Since we know the beacon interval and the monitor timer interval(for now default is 5mins), the total beacon count is calculated in the above manner.